File: ExpenseTracker.Domain/Entities/RecurringExpense.cs
namespace ExpenseTracker.Domain.Entities;

public class RecurringExpense
{
    public int RecurringExpenseId { get; set; }

    public int UserId { get; set; }

    public int CategoryId { get; set; }

    public decimal Amount { get; set; }

    public string? Description { get; set; }

    public string Frequency { get; set; } = string.Empty;

    public DateTime StartDate { get; set; }

    public DateTime NextRunDate { get; set; }

    public DateTime? EndDate { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;

    public Category Category { get; set; } = null!;
}

File: expense-tracker-web/src/types/recurringExpense.ts
export interface RecurringExpense {
  recurringExpenseId: number;
  categoryId: number;
  categoryName: string;
  amount: number;
  description: string | null;
  frequency: string;
  startDate: string;
  nextRunDate: string;
  endDate: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRecurringExpenseRequest {
  categoryId: number;
  amount: number | "";
  description?: string;
  frequency: string;
  startDate: string;
  endDate?: string | null;
}

export interface UpdateRecurringExpenseRequest {
  categoryId: number;
  amount: number;
  description?: string;
  frequency: string;
  endDate?: string | null;
  isActive: boolean;
}